1. Розрахувати необхідний розмір вибірки.\
Ти хочеш провести A/B-тест для двох версій сторінки продукту з очікуваним ефектом у 3% різниці в конверсії.\
Рівень значущості — 5%,\
потужність тесту — 80%.\
Базовий (поточний) рівень конверсії - 10%.\
Використаємо Python, щоб розрахувати необхідний розмір вибірки.

In [ ]:
import numpy as np
import statsmodels.stats.api as sms
from statsmodels.stats.proportion import proportion_effectsize

# Baseline conversion rate (10%)
baseline_conversion = 0.10

# Expected effect (3% relative difference in conversion)
expected_lift = 0.03 * baseline_conversion
variant_conversion = baseline_conversion + expected_lift

# Effect size calculation (Cohen's h)
effect_size = proportion_effectsize(baseline_conversion, variant_conversion)

# Test parameters
alpha = 0.05  # Significance level (5%)
power = 0.80  # Test power (80%)

# Sample size calculation
sample_size = sms.NormalIndPower().solve_power(effect_size=effect_size, power=power, alpha=alpha, ratio=1)

# Output the result
print(f"Required sample size for each group: {int(np.ceil(sample_size))}")


Required sample size for each group: 159059


Отримано результат:\
Required sample size for each group: 159059

**Отриманий результат означає, що потрібно зібрати дані щонайменше від 159059 користувачів для кожної версії сторінки A і B**



2. Оцінити, як зміна рівня значущості вплине на розмір вибірки.\
Ми хочемо побачити, як зміниться необхідний розмір вибірки, якщо ми знизимо рівень значущості до 1%.

In [ ]:
import numpy as np
import statsmodels.stats.api as sms
from statsmodels.stats.proportion import proportion_effectsize

# Baseline conversion rate (10%)
baseline_conversion = 0.10

# Expected effect (3% relative difference in conversion)
expected_lift = 0.03 * baseline_conversion
variant_conversion = baseline_conversion + expected_lift

# Effect size calculation (Cohen's h)
effect_size = proportion_effectsize(baseline_conversion, variant_conversion)

# Test parameters
alpha = 0.01  # Significance level (1%)
power = 0.80  # Test power (80%)

# Sample size calculation
sample_size = sms.NormalIndPower().solve_power(effect_size=effect_size, power=power, alpha=alpha, ratio=1)

# Output the result
print(f"Required sample size for each group: {int(np.ceil(sample_size))}")


Required sample size for each group: 236676


Отримано результат: Required sample size for each group: 236676

**Знизивши рівень значущості до 1% ми побачили, що в такому випадку необхідно зібрати дані щонайменше від 236676 користувачів для кожної версії сторінки A і B. Тобто необхідний розмір вибірки значно збільшився.**

3. Провести розрахунок для меншого ефекту розміру.\
Якщо очікуваний ефект зменшиться до 1% різниці в конверсії, який розмір вибірки знадобиться?


In [3]:
import numpy as np
import statsmodels.stats.api as sms
from statsmodels.stats.proportion import proportion_effectsize

# Baseline conversion rate (10%)
baseline_conversion = 0.10

# Expected effect (1% relative difference in conversion)
expected_lift = 0.01 * baseline_conversion
variant_conversion = baseline_conversion + expected_lift

# Effect size calculation (Cohen's h)
effect_size = proportion_effectsize(baseline_conversion, variant_conversion)

# Test parameters
alpha = 0.01  # Significance level (1%)
power = 0.80  # Test power (80%)

# Sample size calculation
sample_size = sms.NormalIndPower().solve_power(effect_size=effect_size, power=power, alpha=alpha, ratio=1)

# Output the result
print(f"Required sample size for each group: {int(np.ceil(sample_size))}")


Required sample size for each group: 2111540


**Отримано результат: Required sample size for each group: 2111540**

**Якщо очікуваний ефект зменшиться до 1% різниці в конверсії та рівень значущості до 1% то нам знадобиться значно більша вибірка для проведення А/В тестування.**


Розмір вибірки збільшується, коли ми зменшуємо очікувану різницю в конверсії (ефект) і рівень значущості (α).


Чому це відбувається:\
Менший ефект (1% замість 3%) вимагає більшу вибірку для статистичної значущості:

Коли різниця між варіантами (контрольним і тестовим) стає меншою, потрібно більше учасників для того, щоб виявити цей ефект на статистично значущому рівні.

Якщо очікуваний ефект малий (1% різниці), то навіть якщо тест має потужність 80%, його важче виявити через таку маленьку різницю.

Зменшення рівня значущості (α до 1%):

Коли ми встановлюємо більш суворі умови для статистичної значущості (рівень значущості 1%), тест стає більш консервативним. Це означає, що потрібно більше даних, щоб переконатися, що знайдений ефект не є випадковим.

На рівні 1% значущості потрібно більше учасників, щоб мінімізувати ймовірність помилки типу I (false positive) — коли ми помилково відхиляємо нульову гіпотезу.

**Висновок:\
При 1% ефекті та 1% рівні значущості ваша вибірка стає значно більшою — 2,111,540 учасників на кожну групу. Це тому, що ефект дуже малий і рівень значущості високий, тому для виявлення цього ефекту потрібен більший обсяг даних.**
